test/e2e/set/scyllacluster/scyllacluster_shardawareness.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2021 ScyllaDB
+
+package scyllacluster
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/scylladb/scylla-operator/test/e2e/utils"
+
+	"github.com/gocql/gocql"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/scylladb/gocqlx/v2"
+	scyllaclusterfixture "github.com/scylladb/scylla-operator/test/e2e/fixture/scylla"
+	"github.com/scylladb/scylla-operator/test/e2e/framework"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = g.Describe("ScyllaCluster", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewFramework("scyllacluster")
+
+	g.It("should allow to build connection pool using shard aware ports", func() {
+		const (
+			nonShardAwarePort = 9042
+			shardAwarePort    = 19042
+			nrShards          = 2
+		)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		sc := scyllaclusterfixture.BasicScyllaCluster.ReadOrFail()
+		sc.Spec.Datacenter.Racks[0].Members = 1
+
+		// Ensure 2 shards.
+		sc.Spec.Datacenter.Racks[0].Resources.Limits[corev1.ResourceCPU] = resource.MustParse(fmt.Sprintf("%d", nrShards))
```

Comment:
this is harsh.

CI nodes have 2 cores exactly, so the other ~16 tests may keep you out of those CPUs for a long time (== flakes)

also if we ever run a DS by default this won't ever land on a node because of resources

Reply:
it's not going to wait, It still has resource.requests set to low value. Limits in this case only control number of CFQ shares, doesn't affect scheduling.